### PR TITLE
Bug - budi-6076 cannot delete columns in google sheet

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Fixing issue that doesn't allow deleting columns on google spreadsheets datasets.
Changes:
1. Fix UI to not swallow errors
2. Fix error on deletion

Addresses: 
- https://github.com/Budibase/budibase/issues/8575

## App Export
- Nor required

## Screenshots
Tested actions:
1. Add a column, add content
2. Delete the column 
3. Add two more columns, add content
4. Remove the "non-last" column

https://user-images.githubusercontent.com/15987277/221567327-a5d8f5c5-4dbb-4fcf-9bf7-094b27584a9c.mov

## Documentation
- [x] I have reviewed the budibase documentatation to verify if this feature requires any changes. If changes or new docs are required I have written them.

### CODE REVIEW

File: packages/server/src/integrations/tests/googlesheets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        expect(sheet.loadHeaderRow).toBeCalledTimes(1)
+        expect(sheet.setHeaderRow).toBeCalledTimes(1)
+        expect(sheet.setHeaderRow).toBeCalledWith([
+          "name",
+          "description",
+          "location",
+        ])
+
+        // No undefineds are sent
+        expect((sheet.setHeaderRow as any).mock.calls[0][0]).toHaveLength(3)
```
Comment (adrinr) [inline packages/server/src/integrations/tests/googlesheets.spec.ts:118]: This is catching the existing issue, as undefineds were sent as column headers

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 2]
```diff
@@ -192,13 +192,13 @@
...
 
-  function deleteColumn() {
+  async function deleteColumn() {
     try {
       editableColumn.name = deleteColName
       if (editableColumn.name === $tables.selected.primaryDisplay) {
         notifications.error("You cannot delete the display column")
       } else {
-        tables.deleteField(editableColumn)
+        await tables.deleteField(editableColumn)
```
Comment (adrinr) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:201]: This was swallowing the issue in the frontend

**Review by shogunpurple — APPROVED** [review]
LGTM! Nice work

File: packages/server/src/integrations/tests/googlesheets.spec.ts (hunk 1 above)
Comment (mike12345567) [reply packages/server/src/integrations/tests/googlesheets.spec.ts:118]: Nicely done - cool to have test case for this.

**Review by mike12345567 — APPROVED** [review]
Looks great - nice to have test cases in for this too!

**Pull request merged**